addons/website_slides/static/src/js/slides_course_quiz.js

```diff
@@ -210,6 +187,93 @@ odoo.define('website_slides.quiz', function (require) {
             });
         },
 
+        _animateProgressBar: function (self, $modal, rankProgress) {
+            var firstStepXpGainDelay = 750;
+            var secondStepXpGainDelay = 0;
+            var thirdStepXpGainDelay = 750;
+            var oneStepXpGainDelay = firstStepXpGainDelay + secondStepXpGainDelay + thirdStepXpGainDelay;
+            /**
```

Comment:
Those comments are really nice, but I would move them to the method docstring.
inline method comments tend to "lie" when code later changes and people forget to read and move the comments.

You could do something like:

```
/**
* Handles the animation of the karma gain in the following steps:
* 1. Initiate the tooltip which will display...
* 2. Animate the tooltip text...
* 3a. The user does not level up
*     I. When the user doesn't level up the progress bar sim...
* 3b. The user levels up
*     I. This is the first step of ...
*     II.  The second step of the level up animation ...
*     ...
* @private
*/
```
As a bonus, this is included in the generated JS doc.